### ISSUE TITLE
Stack header not announced by accessibility as a Heading on Android

### ISSUE THREAD

**Issue by justin-unterreiner-lulus** [react-navigation/react-navigation#11822]
### Current behavior

The header label for a stack navigator is announced by VoiceOver on iOS as `Home, heading`. But using TalkBack on Android it is just announced as `Home`. I would expect it to be announced in a same or similar way on Android.

This is reproducible using the code from the `Creating a native stack navigator` section of `Hello React Navigation`:
https://reactnavigation.org/docs/hello-react-navigation/#creating-a-native-stack-navigator

Here is a Snack that is produced when clicking the `Try this example on Snack` link.
https://snack.expo.dev/@justin.unterreiner.lulus/react-navigation-talkback-android-title-bug

Reproduced using Expo Go and the above snack.
|iOS|Android|
|---|---|
|![iOS](https://github.com/react-navigation/react-navigation/assets/105250076/a6e10571-2f44-40aa-afd4-7f5bf822789f)|![Android](https://github.com/react-navigation/react-navigation/assets/105250076/faf28492-d8aa-4e02-9a18-b66cdf5a6b62)|

### Expected behavior

TalkBack should announce the label as a Heading similar to VoiceOver on iOS: `Home, Heading`.

### Reproduction

https://snack.expo.dev/@justin.unterreiner.lulus/react-navigation-talkback-android-title-bug

### Platform

- [X] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [X] @react-navigation/stack
- [X] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |6.0.10
| @react-navigation/bottom-tabs          |6.3.1
| @react-navigation/drawer               |6.4.1
| @react-navigation/material-top-tabs    |6.2.1
| @react-navigation/stack                |6.2.1
| @react-navigation/native-stack         |6.6.1
| react-native-safe-area-context         |4.8.2
| react-native-screens                   |~3.29.0
| react-native-gesture-handler           |~2.14.0
| react-native-reanimated                |~3.6.2
| react-native-tab-view                  |^3.0.0
| react-native-pager-view                |6.2.3

**Comment by satya164** [comment]
> I would expect it to be announced in a same or similar way on Android

Android and iOS are different OS. You can't expect them to work the same.

We use the native header, so whatever accessibility is provided is how native components behave. We don't control any accessibility related options for the header.

**Comment by justin-unterreiner-lulus** [comment]
> We use the native header, so whatever accessibility is provided is how native components behave. We don't control any accessibility related options for the header.

Who does control the accessibility options for the native header?

Normally this can be achieved by using the `accessibilityRole="header"` property. This is how `<Text>` works for example.

When you say "native" do you mean that there is not an underlying `<Text>` element in this case?

**Comment by justin-unterreiner-lulus** [comment]
I found a workaround after digging around in the `@react-navigation/native-stack` package source in `HeaderConfig.tsx`.

When the platform is Android, there is a code section that handles applying different header views based on config properties. If either `headerTitle` is a function _or_ `headerTitleAlign` is set to `center` then a `<HeaderTitle>` component will be used instead of the (presumably) "native" fallback:

https://github.com/react-navigation/react-navigation/blob/8367758824590b9affd07ce651f8020aca2a88f6/packages/native-stack/src/views/HeaderConfig.tsx#L228-L262

In looking at the `<HeaderTitle>` component, you can see that it sets the `accessibilityRole="header"` property, which resolves the issue for my use case:

https://github.com/react-navigation/react-navigation/blob/8367758824590b9affd07ce651f8020aca2a88f6/packages/elements/src/Header/HeaderTitle.tsx#L23

So the tradeoff here is that we need to set either a header title render function or set the alignment to center on all screens in order to get the proper accessibility role applied to the header text.
